### PR TITLE
Bump NodeJS to v14 LTS

### LINKED ISSUE #2227: cli: fix wrtc bugs biting us
[wrtc](https://github.com/node-webrtc/node-webrtc) is a NodeJS native module providing WebRTC on this environment. It's very fast (written in C++) and provides the same interface as browser's WebRTC objects, allowing us have a single codebase supporting both environments. There're though some bugs on this module which can become a bigger issue if `raiden-cli` is intended to be a major client on servers, and we need them fixed.

Most of the times, WebRTC data channels can be established and remain like that for a long time, but on cli + wrtc, sometimes it disconnects and can't get connected again. Also, this seems to take down all other WebRTC connections on that node as well, and the process starts consuming 100% of one CPU core.

Fortunately, [we managed to detect this state](https://github.com/raiden-network/light-client/blob/48da3d7b6f946942f29bd7d5c7f3252847a2e459/raiden-ts/src/transport/epics/webrtc.ts#L358-L365) and discard the channel (`connectionState` becomes `failed`), falling back to matrix messages (i.e. the client and protocol continue usable, although wasting one core of CPU/power), so the client remains responsive, but it's a pity to lose WebRTC randomly for the rest of a session.

The entry-point (and possibly a cause or lead to the cause) can be seen as a segfault [if we try to close the connection](https://github.com/raiden-network/light-client/blob/48da3d7b6f946942f29bd7d5c7f3252847a2e459/raiden-ts/src/transport/epics/webrtc.ts#L406-L410). Maybe the 100% CPU issue is because the datachannel gets closed but the peer connection doesn't.

This seems to be more commonly triggered on high-concurrency scenario, like BF6, or a longer-running one, like BF1. When you see a `node` process starting consuming 100% of one core, you can check the logs and see a `RTC: connection failed` log and the retry loop for the webRTC kicking in (and looping each 5-60s, failing and retrying).

We should try to fix this bug ourselves upstream, pay for someone to fix it, or put a bounty out to get it fixed.

Some possibly related upstream bugs:
- https://github.com/node-webrtc/node-webrtc/issues/576
- https://github.com/node-webrtc/node-webrtc/issues/458
- https://github.com/node-webrtc/node-webrtc/issues/636
- https://github.com/node-webrtc/node-webrtc/issues/645

## Steps to Reproduce
1. Run BF1 and/or BF6 scenarios, check logs when/if one `node` process starts consuming 100% CPU.
2. Possibly related to the same bug, or a different one: uncomment that `connection.close()` line, get a client connected via webRTC with another (e.g. by having 2 raiden-cli clients having a Raiden channel between them and being online at the same time), terminates one of them, and see the other segfaults when closing the connection. Usually the terminated one too segfaults on `connection.close()` while shutting down.

## Expected Result
- webRTC channels can get reliably connected, closed, retried and reconnected during a session

## Actual Result
- `connection failed` errors happen and make webRTC unusable for the rest of that session's lifecycle.

## Remark
- After this has been fixed revert the ForceExit in the E2E nightly tests (see https://github.com/raiden-network/light-client/issues/2262)

### LINKED ISSUE #2262: Remove forceExit flag for E2E test of the SDK in CircleCI
## Description
To fix #2242 we introduced a temporally fix with #2261 that is "hiding" this error to make the CI pipeline working again.
This can only happen when the origin issue with #2227 has been resolved.

## Acceptance criteria
- WebRTC bug was fixed as requirement
- The `--forceExit` got removed at the `test_sdk_e2e` job at CircleCI

## Tasks
- [ ]

### LINKED ISSUE #2311: Upgrade infrastructure to Node 14
Upgrade infrastructure to Node 14

### PR DESCRIPTION
**Pull request by andrevmatos**
Fixes #2311 
~~Fixes `#2227`~~
Fixes #2262 

**Short description**
Also:
- Some fixes for proper `yarn` cache management on CircleCI
- Call `close()` on [WebRTC connection](https://github.com/raiden-network/light-client/blob/dde6594a55c4cb14e0268c9e9493de09404682a8/raiden-ts/src/transport/epics/webrtc.ts#L415-L416); ensures the connection is properly teared down, as `wrtc` doesn't segfault on close anymore on Node v14, allowing proper teardown of RTC channels
  - Edit: the apparent infinite loop bug on `wrtc` is still present on this branch/node14, so #2227 isn't solved, but this improves  it and specially the connection teardown phase
- It was noticed the e2e tests were trying to shutdown before last db flush, which made it fail even though it could exit properly; to fix this, `Raiden.stop` was made asynchronous, and resolves after `db` is properly flushed and gracefully closed;
  - This shouldn't be needed on browsers, they're fast enough; only on `memory` adapter on integration/e2e tests this could be seen
- With both issues above fixed, we can now have proper teardown of e2e SDK tests; there were some infra issues, which were also addressed on this PR: now, e2e tests complete and exit cleanly! :tada: 

image

**Definition of Done**

- [x] Steps to manually test the change have been documented
- [x] Acceptance criteria are met
- [ ] Change has been manually tested by the reviewer (dApp) 

**Steps to manually test the change (dApp)**

1.
2.

### CODE REVIEW

**Comment by andrevmatos** [comment]
I've built the new image successfully here. I'll push the upgradeto Docker Hub as soon as this gets merged.

File: .circleci/config.yml [hunk 1]
```diff
@@ -76,11 +71,17 @@ jobs:
     working_directory: *working_dir
     steps:
       - attach_workspace: *attach_options
-      - restore_cache: *package_lock_key
+      - restore_cache:
+          name: Restore Yarn Package Cache
+          keys:
+            - raiden-v1-yarn-{{ checksum "yarn.lock" }}
```
Comment (palango) [inline .circleci/config.yml:77]: The old `cache_key` had a `.Branch` in it as well. might make sense to keep that.

File: e2e-environment/run-e2e-tests.sh [hunk 2]
```diff
@@ -27,6 +27,8 @@ docker run --detach --rm \
   raidennetwork/lightclient-e2e-environment \
   >/dev/null
 
+sleep 5
```
Comment (palango) [inline e2e-environment/run-e2e-tests.sh:30]: Why is this needed?

**Review by palango — APPROVED** [review]
Nice PR. Left two nits - handle at your discretion.

File: .circleci/config.yml (hunk 1 above)
Comment (andrevmatos) [reply .circleci/config.yml:77]: IMO, there's no point into caching different branches differently. Even because usually different PRs would be seen as different branches. If the `yarn.lock` matches (and actually even if don't), we want the yarn cache to be... _cached_, so it speeds up the install for new runs. Not sure why `.Branch` was there before, but I fail to see a reason for that.

File: e2e-environment/run-e2e-tests.sh (hunk 2 above)
Comment (andrevmatos) [reply e2e-environment/run-e2e-tests.sh:30]: On local runs, I've seen the LCs starting too quickly after the container got started, and failed because it'd not be able to hit matrix/synapse in a usable state. Giving some time for the containers to start did fix it for me.
